src/bidiMapper/scripts/eval.es

```diff
@@ -0,0 +1,115 @@
+(() => {
+  class Serializer {
+    _idToObject = new Map();
+    _objectToId = new WeakMap();
+
+    constructor() {}
+
+    serialize(value) {
+      if (value === undefined) {
+        return { type: 'undefined' };
+      } else if (value === null) {
+        return { type: 'null' };
+      } else if (typeof value === 'string') {
+        return { type: 'string', value };
+      } else if (typeof value === 'number') {
+        let serialized;
+        if (Number.isNaN(value)) {
+          serialized = 'NaN';
+        } else if (value === 0 && 1 / value === -Infinity) {
+          serialized = '-0';
+        } else if (value === -Infinity) {
+          serialized = '-Infinity';
+        } else if (value === +Infinity) {
+          serialized = '+Infinity';
+        } else {
+          serialized = value;
+        }
+        return { type: 'number', value: serialized };
+      } else if (typeof value === 'boolean') {
+        return { type: 'boolean', value };
+      } else if (value instanceof Object) {
+        // TODO: Recursive serialize.
+        let id = this._objectToId.get(value);
+        if (!id) {
+          id = uuid();
+          this._objectToId.set(value, id);
+          this._idToObject.set(id, new WeakRef(value));
+        }
+      } else {
+        throw new Error('not yet implemented');
+      }
+      return { type: 'object', objectId: uuid() };
+    }
```

Comment:
```suggestion
        return { type: 'object', objectId: id };
      }
      throw new Error('not yet implemented');
    }
```
otherwise the `objectId` is never returned